src/RequestFactory.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+namespace GuzzleHttp;
+
+use Psr\Http\Message\RequestInterface;
+
+class RequestFactory implements RequestFactoryInterface
+{
+
+    public static function create($method, $uri = '', array $options = [])
```

Comment:
Given that this factory will likely only be used in the context of a client, this helper method doesn't seem necessary.

Reply:
My use case is creating Request instances to be sent later, so I don't have a client when I'm creating them.
This also provides a "safe" creation method that doesn't involve references